core/src/events/LambdaEventReceiver.cpp

```diff
@@ -44,5 +47,27 @@ namespace ledger {
         void LambdaEventReceiver::onEvent(const std::shared_ptr<api::Event> &event) {
             _function(event);
         }
+
+        std::shared_ptr<LambdaEventReceiver> make_promise_receiver(
+                Promise<Unit>& promise,
+                const std::vector<api::EventCode> &successCodes,
+                const std::vector<api::EventCode> &failureCodes
+        ) {
+            return make_receiver([=] (const std::shared_ptr<api::Event> &event) mutable {
```

Comment:
It "transforms" a promise to a receiver. I capture by copy because it is an asynchronous interface, the succesCodes and failureCodes vector won't be alive when I query them otherwise.